skglm/experimental/quantile_huber.py

```diff
@@ -0,0 +1,206 @@
+
+    def __init__(self, quantile=0.5, alpha=0.1, delta_init=1.0, delta_final=1e-3,
+                 n_deltas=10, max_iter=1000, tol=1e-4, verbose=False,
+                 solver="AndersonCD", fit_intercept=True):
+        self.quantile = quantile
+        self.alpha = alpha
+        self.delta_init = delta_init
+        self.delta_final = delta_final
+        self.n_deltas = n_deltas
+        self.max_iter = max_iter
+        self.tol = tol
+        self.verbose = verbose
+        self.solver = solver
+        self.fit_intercept = fit_intercept
+
+    def fit(self, X, y):
+        """Fit using progressive smoothing: delta_init --> delta_final."""
+        w = np.zeros(X.shape[1])
+        deltas = np.geomspace(self.delta_init, self.delta_final, self.n_deltas)
+
+        if self.verbose:
+            print(
+                f"Progressive smoothing: delta {self.delta_init:.3f} --> "
+                f"{self.delta_final:.3f} in {self.n_deltas} steps")
+
+        datafit = QuantileHuber(quantile=self.quantile, delta=self.delta_init)
+        penalty = L1(alpha=self.alpha)
+        # Solver selection
+        if isinstance(self.solver, str):
+            if self.solver == "FISTA":
+                if self.fit_intercept:
+                    import warnings
+                    warnings.warn(
+                        "FISTA solver does not support intercept. "
+                        "Falling back to fit_intercept=False."
+                    )
+                    self.fit_intercept = False
+                solver = FISTA(max_iter=self.max_iter, tol=self.tol)
+                solver.warm_start = True
+            elif self.solver == "AndersonCD":
+                solver = AndersonCD(max_iter=self.max_iter, tol=self.tol,
+                                    warm_start=True, fit_intercept=self.fit_intercept)
+            else:
+                raise ValueError(f"Unknown solver: {self.solver}")
+        else:
+            solver = self.solver
+
+        est = GeneralizedLinearEstimator(
+            datafit=datafit, penalty=penalty, solver=solver)
+
+        for i, delta in enumerate(deltas):
+            datafit.delta = float(delta)
+
+            est.fit(X, y)
+            w = est.coef_.copy()
+
+            if self.verbose:
+                residuals = y - X @ w
+                if self.fit_intercept:
+                    residuals -= est.intercept_
+                coverage = np.mean(residuals <= 0)
+                pinball_loss = np.mean(residuals * (self.quantile - (residuals < 0)))
+
+                print(
+                    f"  Stage {i+1:2d}: delta={delta:.4f}, "
+                    f"coverage={coverage:.3f}, pinball_loss={pinball_loss:.6f}, "
+                    f"n_iter={est.n_iter_}"
+                )
+
+        self.est = est
+
+        return self
+
+    def predict(self, X):
+        """Predict using the fitted model."""
+        if not hasattr(self, "est"):
+            raise ValueError("Call 'fit' before 'predict'.")
```

Comment:
copy the error message that is printed when you predict with LinearRegression() is sklearn without fitting yet

and raise a `NotFittedError`